monkey/agent_plugins/exploiters/rdp/src/in_memory_file_provider.py

```diff
@@ -0,0 +1,37 @@
+from typing import Dict, Optional
+
+from aardwolf.extensions.RDPECLIP.fileprovider import FileProvider
+from egg_timer import EggTimer
+
+
+class InMemoryFileProvider(FileProvider):
+    """
+    Provides data from files stored in memory to the clipboard
+    """
+
+    def __init__(
+        self,
+        files: Optional[Dict[str, bytes]] = None,
+        timer: Optional[EggTimer] = None,
+        timeout: float = 5,
+    ):
```

Comment:
NIT: I think the order of optional parameters here isn't very intuitive. I would recommend either

```suggestion
    def __init__(
        self,
        files: Optional[Dict[str, bytes]] = None,
        timeout: float = 5,
        timer: Optional[EggTimer] = None,
    ):
```

or 

```suggestion
    def __init__(
        self,
        *,
        files: Optional[Dict[str, bytes]] = None,
        timer: Optional[EggTimer] = None,
        timeout: float = 5,
    ):
```